eng/scripts/Sdk-Generate.ps1

```diff
@@ -0,0 +1,144 @@
+param(
+  [string] $InputJsonPath = $GenerateInput,
+  [string] $OutputJsonPath = $GenerateOutput,
+  [string] $RepoRoot = "${PSScriptRoot}/../.."
+)
+
+$inputFileContent = Get-Content $InputJsonPath | ConvertFrom-Json
+$inputFilePaths = $inputFileContent.changedFiles;
+$inputFilePaths += $inputFileContent.relatedReadmeMdFiles;
+$inputFilePaths = $inputFilePaths | Select-Object -Unique
+
+$changedFilePaths = $inputFilePaths -join "`n";
+Write-Host "List of changed swagger files and related readmes:`n$changedFilePaths`n"
+
+$autorestFilesPath = Get-ChildItem -Path "$RepoRoot/sdk"  -Filter autorest.md -Recurse | Resolve-Path -Relative
+
+Write-Host "Updating autorest.md files for all the changed swaggers."
+$sdksInfo = @{}
+$headSha = $inputFileContent.headSha
+$repoHttpsUrl = $inputFileContent.repoHttpsUrl
+foreach ($inputFilePath in $inputFilePaths) {
+  foreach ($path in $autorestFilesPath) {
+    $fileContent = Get-Content $path
+    $isUpdatedLines = $false
+    $updatedLines = foreach ($line in $fileContent) {
+      $regexForMatchingShaAndPath = "[\/][0-9a-f]{4,40}[\/]$inputFilePath"
+      if ($line -match $regexForMatchingShaAndPath) {
+        $line -replace "https:\/\/[^`"]*$regexForMatchingShaAndPath", "$repoHttpsUrl/blob/$headSha/$inputFilePath"
+
+        $isUpdatedLines = $true
+        $sdkpath = (get-item $path).Directory.Parent.FullName | Resolve-Path -Relative
+        if (!$sdksInfo.ContainsKey($sdkpath)) {
+          $specReadmePath = $inputFileContent.relatedReadmeMdFiles -match $inputFilePath
+          $sdksInfo.Add($sdkpath, $specReadmePath)
+        }
+      }
+      else {
+        $line
+      }
+    }
+    if ($isUpdatedLines) {
+      $updatedLines | Out-File -FilePath $path
+    }
+  }
+}
+
+Write-Host "Updated autorest.md files for all the changed swaggers. `n"
+
+function Test-PreviousScript() {
+  $result = $null
+  if ($LASTEXITCODE -eq 0) {
+    $result = "succeeded"
+  }
+  else {
+    $result = "failed"
+  }
+  Write-Host "Result: $result"
+  return $result
+}
+
+$packages = @()
+foreach ($sdkPath in $sdksInfo.Keys) {
+  $packageName = Split-Path $sdkPath -Leaf
+  Write-Host "Generating code for " $packageName
+  $srcPath = Join-Path $sdkPath 'src'
+  dotnet msbuild /restore /t:GenerateCode $srcPath
+
+  $artifacts = @()
+  $hasBreakingChange = $null
+  $content = $null
```

Comment:
Tip: this isn't necessary in PowerShell. Variables can be created at any time and are the same as $null if not assigned by the time you read them. They are always available in the scope in which they were declared or assigned. Statements like `if` below don't create new scopes, but scriptblocks (like passed to `foreach-object`) will and then you have to use the `$script:` prefix on a variable to set the parent scope's variables anyway (whether declared or not). See `help about_scopes` for details.

Reply:
I'm setting these null because if it fails while generating code/creating artifacts [here](https://github.com/Azure/azure-sdk-for-net/blob/c5ef13f487eb3190b8d46be488470b17e1d35aee/eng/scripts/Automation-Sdk-Generate.ps1#L89-L97), it will go to the else block where I'm not setting values for  `$hasBreakingChange` and `$content`, hence it will have previous values(that we don't want to use for the current package). If it fails while generating code/creating artifacts, I want to set null values.

Reply:
Depending on the strict mode (https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/set-strictmode?view=powershell-7.1) you might actually need to initialize these values.  I've personally started setting Strict mode version 3 in my scripts to help catch stupid issues with uninitialized variables where I had a typo or something.